src/coord/src/coord.rs

```diff
@@ -802,11 +805,13 @@ impl Coordinator {
             );
         }
 
-        self.dataflow_client
-            .storage_mut()
-            .advance_all_table_timestamps(advance_to)
-            .await
-            .unwrap();
+        for (id, updates) in self.volatile_wal.drain(..) {
+            self.dataflow_client
+                .storage_mut()
+                .append(id, updates, advance_to)
+                .await
+                .unwrap();
+        }
```

Comment:
Quick skim, but I think this charge risks *not advancing* timestamps of tables that do not undergo changes. Someone somewhere wants to enumerate "all" tables and append "nothing" into them. This has been imo a bit of debt we've had for a while (that `coord.rs` does not need to state table ids, and relies on `dataflow` to know what are all the current tables) so .. it may be gross to unpick. Lmk if discussing would help!

Reply:
Sounds great! It's a bit crazy that we are scanning all catalog entries; maybe a `// TODO` to tidy some of it up in the future (for ADAPTER, not you)?

Reply:
fixed